### DISCUSSION TITLE
How to use QueryBuilder in pattern with repositories and services?

### DISCUSSION THREAD

**Discussion by Georgiiagon** [spatie/laravel-query-builder#538] [Q&A]
In controllers, we should use services instead of repositories or models.
In services, we should use repositories instead of models. 
The whole business logic is described in services as well.

Should I create a separate "repository" for QueryBuilder or use the same as for the model?

**Comment by vitalijalbu** [comment]
> In controllers, we should use services instead of repositories or models. In services, we should use repositories instead of models. The whole business logic is described in services as well.
> 
> Should I create a separate "repository" for QueryBuilder or use the same as for the model?

Hi same question i had, do you found out a professional solution?
